diagnostic_updater/src/example.cpp

```diff
@@ -36,6 +36,11 @@
 #include <std_msgs/Bool.h>
 #include <diagnostic_updater/publisher.h>
 
+// ERROR defined in windows.h causes name collision, undefine the macro to fix the issue
+#ifdef ERROR
+#undef ERROR
+#endif
```

Comment:
This may work here, but it indicates a problem with the message definition, and every other user of these messages will have the same issue. Is there some other way to prevent this, in a way that would apply to all of the users of this library?

Reply:
another method that might sound better and more universal (but actually bad) is to move similar macro guard into where `.msg` is generated into `.h`. however, there are 2 problems with this:
- the (centralized) generator would have no idea of the names that each `.msg` would use, it can never prevent all possible name collisions
- if all similar macros are un-defined, it might be doing extra unnecessary work, and actually break code that uses such macros

based on these, the very likely best (at this point) way is still just to handle these special cases right at where it happens =)

Reply:
ROS2 is (at this point) recommending similar fix to this problem, https://github.com/ros2/common_interfaces/issues/44#issuecomment-397516076

(thanks to @seanyen-msft for finding this)

Reply:
Ok.